Updating the readme to reflect the Atlas Device SDK branding (#6448)
realm/realm-js: packages/realm/README.md

> [!NOTE]
> Realm is now Atlas Device SDK - [Learn More](https://www.mongodb.com/blog/post/realm-now-part-atlas-platform?utm_medium=github_atlas_CTA&utm_source=realm_js_github)

<img align="right" style="min-width:150px;width:30%;" src="https://raw.githubusercontent.com/realm/realm-js/main/media/atlas-device-sync.svg" alt="Atlas Device Sync">

> # Atlas Device SDK

Author: I was contemplating putting "(for JavaScript & TypeScript)" here as well, but it felt a bit verbose in a title.

> Atlas Device SDK is a suite of app development tools optimized for data access and persistence on mobile and edge devices. Use the SDKs to build data-driven mobile, edge, web, desktop, and IoT apps.

Reviewer: What docs has been told and what we're working with is that Realm is the device data persistence layer part of Atlas Device SDK. Check out the Introduction page in the docs: https://www.mongodb.com/docs/realm/introduction/

This project hosts the JavaScript / TypeScript versions of the SDK. Currently, we support React Native (JSC & Hermes on iOS & Android), Node.js and Electron (on Windows, MacOS and Linux).

> An integral part of this SDK is a mobile database that runs directly inside phones, tablets or wearables. This used to be called Realm and is still referred to as Realm throughout our codebase as we're currently focused on improving the experience over renaming everything, causing lots of breaking changes, etc.

Author: Okay. That makes sense. But then I definitely think our announcing post should be clarified to reflect that nuance: https://www.mongodb.com/blog/post/realm-now-part-atlas-platform

> An integral part of this SDK is a mobile database that runs directly inside phones, tablets or wearables. This used to be called Realm and is still referred to as Realm throughout our codebase as we're currently focused on improving the experience over renaming everything, causing lots of breaking changes, etc.

Author: It's basically not Realm (the database) but the client for what was formerly know as Stitch which has a new name then.

> [!NOTE]
> Realm is now Atlas Device SDK - [Learn More](https://www.mongodb.com/blog/post/realm-now-part-atlas-platform?utm_medium=github_atlas_CTA&utm_source=realm_js_github)

<picture>
    <source srcset="https://raw.githubusercontent.com/realm/realm-js/main/media/logo-dark.svg" media="(prefers-color-scheme: dark)" alt="realm by MongoDB">
    <img src="https://raw.githubusercontent.com/realm/realm-js/main/media/logo.svg" alt="realm by MongoDB">
</picture>

> # The Realm database

Reviewer 2: ```suggestion
# The Realm Database
```

Realm is a mobile database that runs directly inside phones, tablets or wearables. This project hosts the JavaScript & TypeScript implementation of Realm. Currently, we support React Native (JSC & Hermes on iOS & Android), Node.js and Electron (on Windows, MacOS and Linux).

> ## What is the Atlas Device SDKs?

Reviewer 2: ```suggestion
## What are the Atlas Device SDKs?
```

<img align="right" style="min-width:150px;width:20%;" src="https://raw.githubusercontent.com/realm/realm-js/main/media/atlas-device-sync.svg" alt="Atlas Device Sync">

> The [Atlas Device SDKs](https://www.mongodb.com/docs/realm/sdk/) are a connection of language & platform specific SDKs, each with a suite of app development tools optimized for data access and persistence on mobile and edge devices. Use the SDKs to build data-driven mobile, edge, web, desktop, and IoT apps.

Reviewer 2: ```suggestion
The [Atlas Device SDKs](https://www.mongodb.com/docs/realm/sdk/) are a connection of language and platform specific SDKs, each with a suite of app development tools optimized for data access and persistence on mobile and edge devices. Use the SDKs to build data-driven mobile, edge, web, desktop, and IoT apps.
```

* [`realm`](https://www.npmjs.com/package/realm) (NPM tag: `latest`) on node.js (active LTS) and Electron on Windows, MacOS, and Linux
* [`realm`](https://www.npmjs.com/package/realm) (NPM tag: `latest`) with the latest React Native version 0.71.0 on Android and iOS
* [`@realm/react`](https://www.npmjs.com/package/@realm/react) (NPM tag: `latest`) in conjunction with the latest [`realm`](https://www.npmjs.com/package/realm) release

### Tier 2 - best effort

Some packages are considered to be mature and stable, and we will support them as good as we can when time permits.

* [`realm-web`](https://www.npmjs.com/package/realm-web) (NPM tag: `latest`)
* [Realm Studio](https://github.com/realm/realm-studio) ([latest](https://github.com/realm/realm-studio/releases/latest) release) on Windows, MacOS, and Linux
* Any other release of [`realm`](https://www.npmjs.com/package/realm) not covered by tier 1

### Tier 3 - experimental

The third tier covers our experimental packages. We work on them occasionally, and they are likely to change radically when we do.

* [`@realm/babel-plugin`](https://www.npmjs.com/package/@realm/babel-plugin)

> * [`realm-flipper-plugin`](https://www.npmjs.com/package/realm-flipper-plugin)

Reviewer 3: I think we should remove it

Author: ```suggestion
```

Author: I'll leave actually removing it from the repository, to a separate PR 🙂

> ## Documentation

Reviewer 2: In the first paragraph here it still says "Realm React Native SDK" and "Realm Node.js SDK". (Couldn't comment on the exact lines.)

Reviewer: And the heading for this section has the old names too.
"Atlas Device SDK for React Native and Atlas Device SDK for Node.js" as a heading is a bit much though. We could probably simplify to "Atlas Device SDKs for React Native and Node.js".

Author: Updated 👍

### Integration with React Native

> Realm is a general SDK which provide you persistence of objects and the capability to perform advanced queries on the objects. You can have tighter integration with React Native by using [`@realm/react`](https://www.npmjs.com/package/@realm/react).

Reviewer 2: ```suggestion
Realm is a general SDK which provides persistence of objects and the capability to perform advanced queries on the objects. You can have easier integration with React Native by using [`@realm/react`](https://www.npmjs.com/package/@realm/react).
```

> Realm is a general SDK which provide you persistence of objects and the capability to perform advanced queries on the objects. You can have tighter integration with React Native by using [`@realm/react`](https://www.npmjs.com/package/@realm/react).

Reviewer: ```suggestion
The Atlas Device SDK for React Native provides you with object persistence and
advanced queries for persisted objects. You can have tighter integration with React Native by using [`@realm/react`](https://www.npmjs.com/package/@realm/react).
```
